CMSgov/mdct-mcr
# add error boundary for unhandled errors

## Description

We need to wrap portions of the app in an error boundary to handle any unhandled errors that may occur. 

Changes:
- Add `react-error-boundary` dependency
- Add resolvers for `@types/react` and `@types/react-dom` because of CMSDS implicit `*` dependency. See [more here](https://github.com/facebook/react/issues/24304).
- Make error view for a fallback display on unhandled error.
- Minor styling tweak to ErrorAlert.

Along with earlier error handling PR, satisfies [MDCT-259: Error boundary & Alert component](https://qmacbis.atlassian.net/browse/MDCT-259)

<img width="1271" alt="Screen Shot 2022-06-15 at 11 39 07" src="https://user-images.githubusercontent.com/27705928/174082751-074853e6-f1ab-463b-89b7-9a8db089c262.png">

### How to test

1. Pull and run locally
2. Make an error throw in the app
    - Open `Home.tsx`
    - Make a new line on line 17
    - Paste this code: `throw new Error()` and save
4. App should now throw an error and display a nice error page for you.

### Changed Dependencies

- Need to run `yarn install` to get new dependency `react-error-boundary`.

## Code author checklist

## Reviewer checklist (two different people)
